src/basic_block.rs

```diff
@@ -424,7 +424,19 @@ impl<'ctx> BasicBlock<'ctx> {
     /// Set name of the `BasicBlock`.
     pub fn set_name(&self, name: &str) {
         let c_string = to_c_str(name);
-        unsafe { LLVMSetValueName(LLVMBasicBlockAsValue(self.basic_block), c_string.as_ptr()) };
+
+        #[cfg(any(feature = "llvm4-0", feature = "llvm5-0", feature = "llvm6-0"))]
+        {
+            use llvm_sys::core::LLVMSetValueName;
+
+            unsafe { LLVMSetValueName(LLVMBasicBlockAsValue(self.basic_block), c_string.as_ptr()) };
+        }
+        #[cfg(not(any(feature = "llvm4-0", feature = "llvm5-0", feature = "llvm6-0")))]
+        {
+            use llvm_sys::core::LLVMSetValueName2;
+
+            unsafe { LLVMSetValueName2(LLVMBasicBlockAsValue(self.basic_block), c_string.as_ptr(), name.len()) };
```

Comment:
I suppose we could optimize this a little by moving the to_c_str call into the above block and just use the str len in this block

Reply:
Indeed, done.